packages/framework/src/parser.ts

```diff
@@ -162,3 +168,160 @@ export const unwrapWidget = (obj: Record<string, unknown>): EnsembleWidget => {
     properties: properties as Record<string, unknown>,
   };
 };
+
+const unwrapBody = (
+  viewWidget: EnsembleWidget,
+  header: EnsembleHeaderModel | undefined,
+  footer?: EnsembleFooterModel | undefined,
+): EnsembleWidget => {
+  const marginTop = !header
+    ? "0px"
+    : !header?.styles?.titleBarHeight
+    ? "56px"
+    : typeof header?.styles?.titleBarHeight === "number"
+    ? header?.styles?.titleBarHeight + "px"
+    : header?.styles?.titleBarHeight;
+  const marginBottom = !footer
+    ? "0px"
+    : !("styles" in footer)
+    ? "56px"
+    : typeof footer?.styles?.height === "number"
+    ? footer?.styles?.height + "px"
+    : footer?.styles?.height;
+
+  // default body styles
+  const defaultStyles = {
+    height: `calc(100vh - ${marginTop} - ${marginBottom})`,
+    overflow: "auto",
+    marginTop,
+    marginBottom,
+  };
+
+  return {
+    name: "Column",
+    properties: {
+      children: [viewWidget],
+      styles: defaultStyles,
+    },
```

Comment:
Styling defaults and UI logic should be part of the component, not in the parser.